telematic_system/telematic_apps/web_app/client/src/pages/Login.js

```diff
@@ -13,164 +13,203 @@
-                    userOrgsData.forEach(userOrg => {
-                        if (userOrg.org_id === resData.org_id) {
-                            authContext.updateRole(userOrg.role);
-                        }
-                    });
-                    //Get current user organization name
-                    const org_response = listOrgs();
-                    org_response.then(data => {
-                        if (data !== undefined && Array.isArray(data) && data.length !== 0) {
-                            data.forEach(item => {
-                                if (item !== undefined && item.id !== undefined && parseInt(item.id) === parseInt(resData.org_id)) {
-                                    authContext.updateOrg(resData.org_id, item.name);
-                                }
-                            });
-                        }
-                    }).catch(err => {
-                        console.error(err);
-                        return;
-                    });
+    //Checking if login is successful
+    status
+      .then((resData) => {
+        if (resData.errCode !== undefined) {
+          setLoginErrMsg(resData.errMsg);
+          setLoginState(false);
+          return;
+        }
+        //Checking if user belong to an organization
+        const org_user_response = getOrgsByUser(resData.id);
+        org_user_response
+          .then((userOrgsData) => {
+            if (
+              userOrgsData !== undefined &&
+              Array.isArray(userOrgsData) &&
+              userOrgsData.length > 0
+            ) {
+              //Get Current active organization user role
+              userOrgsData.forEach((userOrg) => {
+                if (userOrg.org_id === resData.org_id) {
+                  authContext.updateRole(userOrg.role);
                 }
-            }).catch(err => {
-                console.error(err);
-                return;
-            });
-            authContext.login(
-                resData.id,
-                resData.login,
-                resData.token,
-                resData.tokenExpiredAt,
-                resData.email,
-                resData.last_seen_at,
-                resData.org_id,
-                resData.org_name,
-                resData.name,
-                resData.is_admin);
-            setLoginState(true);
-        }).catch(error => {
-            console.error(error);
-            setLoginErrMsg('Your username or password is incorrect!');
-            setLoginState(false);
+              });
+              //Get current user organization name
+              const org_response = listOrgs();
+              org_response
+                .then((data) => {
+                  if (
+                    data !== undefined &&
+                    Array.isArray(data) &&
+                    data.length !== 0
+                  ) {
+                    data.forEach((item) => {
+                      if (
```

Comment:
Should we be throwing an error if item.id doesn't match any org_ids?

Reply:
Why would we throw error? What kind of error to throw?